packages/lodestar/src/cache/abstract.ts

```diff
@@ -0,0 +1,68 @@
+import {AnySSZType} from "@chainsafe/ssz-type-schema";
+import {CacheId, CacheIdFunction} from "./types";
+import {clone, hashTreeRoot} from "@chainsafe/ssz";
+
+export abstract class CacheItem<T> {
+
+  protected sszType: AnySSZType;
+
+  protected idFunction: CacheIdFunction<T>;
+
+  protected cache: Map<CacheId, T> = new Map();
+
+  protected constructor(sszType: AnySSZType, idFunction?: CacheIdFunction<T>) {
+    this.sszType = sszType;
+    this.idFunction = idFunction || this.hashRootIdFunction;
```

Comment:
where does `this.hashRootIdFunction` come from? it will always be null?

Reply:
it's defined in class. Although I could move it to separate file with some common id generators